### PR TITLE
Cbusman/filter dropdown component

### PR DESCRIPTION
**Pull request by CarolBusman**
Currently using `iron-pages` to switch between filter categories as `d2l-tabs` is behaving oddly when put inside a `d2l-dropdown`.

### CODE REVIEW

**Comment by dbatiste** [comment]
Does the design call for the use of `d2l-tabs`?  If so, then I think we should figure out what's going on there instead of iron pages.

**Comment by CarolBusman** [comment]
> Does the design call for the use of `d2l-tabs`? If so, then I think we should figure out what's going on there instead of iron pages.

When I was discussing with @svanherk, she brought up that `iron-pages` won't make sense after we move to LitElement since it's a polymer thing, and that using internal components would make things easier.

**Comment by svanherk** [comment]
The "design" we are going for is basically a copy of the `my-courses` filter, while the official design is still fleshed out.  `my-courses` is using the `iron-pages`, and I was hoping we could move away from that and use our version.

**Comment by dbatiste** [comment]
Yeah, totally agree with that.  Many times the iron components have been a quick solution, but hasn't always landed us in the place we want to be.

This idea of tabs in dropdowns has come up a few times, and envisioned that it could eventually be a component/feature with dropdowns.

File: components/d2l-filter-dropdown/d2l-filter-dropdown-localize-behavior.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	 * Localization resources.
+	 */
+		resources: {
+			value: function() {
+				return {
+					'ar': {
+						clear: 'مسح',
+						filter: 'التصفية',
+						filterBy: 'تصفية بحسب',
+						filterMultiple: 'Filter: {numOptions} Filters',
```
[the remark this answers is not in the document]
Comment (CarolBusman) [reply components/d2l-filter-dropdown/d2l-filter-dropdown-localize-behavior.js]: All the translations were taken from my courses and this one does not have a translation: https://github.com/Brightspace/d2l-my-courses-ui/blob/master/lang/ar.json#L20.

File: components/d2l-filter-dropdown/d2l-filter-dropdown-page.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+				value: ''
+			},
+			parentTitle: {
+				type: String,
+				value: ''
+			},
+			options: {
+				type: Array,
+				value: [
+					// {
```
[the remark this answers is not in the document]
Comment (CarolBusman) [reply components/d2l-filter-dropdown/d2l-filter-dropdown-page.js]: I left these in on purpose to show what the object structure looks like. Should they be removed?

[the remark this answers is not in the document]
Comment (AdmiralCoco) [reply components/d2l-filter-dropdown/d2l-filter-dropdown-page.js]: Ah got it, I thought that might be the case - sounds good

**Comment by svanherk** [comment]
We're looking to merge this in now, so we can test it with the HM wrapper.  I still have a few follow-up fixes to some defects, so feel free to still review and I will make those changes in the upcoming PRs!

**Pull request merged**